### PR TITLE
Update master to Pastas 1.4

### PR DESCRIPTION
**Pull request by raoulcollenteur**
# Short Description
This PR updates the master branch to Pastas 1.4. Many small updates and added python 3.12 support. Signatures module received a big update. Pastas is now compatible with Pandas 2.2 as well.

# Checklist before PR can be merged:
- [x] is documented
- [x] Format code with [Black formatting](https://black.readthedocs.io)
- [x] type hints for functions and methods
- [x] tests added / passed
- [x] Example Notebook (for new features)
- [x] Remove output for all notebooks with changes

### CODE REVIEW

**Comment by martinvonk** [comment]
I see that the test_notebooks.py is tested by the py38-py312 test suites while this is not intended. Only the _Test suite for all unit tests including Notebooks_ should do this. It worked in the past; with the `@pytst.mark.notebooks` decorator..

File: pastas/version.py
Comment (martinvonk) [inline pastas/version.py:1]: remove `b`

File: pastas/version.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 logger = logging.getLogger(__name__)
 
-__version__ = "1.3.0"
+__version__ = "1.4.0b"
```
Comment (martinvonk) [inline pastas/version.py:7]: remove `b`

**Comment by martinvonk** [comment]
I also get this warning:

> This branch is out-of-date with the base branch
> Merge the latest changes from `master` into this branch.

**Review by martinvonk — APPROVED** [review]

**Comment by raoulcollenteur** [comment]
Will merge the PR after all the checks have passed and release Pastas 1.4!

Thanks for all the work from everyone!

Cheers,
Raoul

**Review by dbrakenhoff — APPROVED** [review]
Looking good!

**Pull request merged**